controllers/storageoscluster/metrics-exporter_operand.go

```diff
@@ -100,15 +122,32 @@ func (c *MetricsExporterOperand) Ensure(ctx context.Context, obj client.Object,
 	ctx, span, _, log := instrumentation.Start(ctx, "MetricsExporterOperand.Ensure")
 	defer span.End()
 
-	b, err := getMetricsExporterBuilder(c.fs, obj, c.kubectlClient)
-	if err != nil {
+	cluster, ok := obj.(*storageoscomv1.StorageOSCluster)
+	if !ok {
+		err := fmt.Errorf("failed to convert %v to StorageOSCluster", obj)
 		span.RecordError(err)
 		return nil, err
 	}
 
-	cluster, ok := obj.(*storageoscomv1.StorageOSCluster)
-	if !ok {
-		err := fmt.Errorf("failed to convert %v to StorageOSCluster", obj)
+	var (
+		cm  *corev1.ConfigMap
+		err error
+		cfg *metricsv1.MetricsExporterConfig
+	)
+	if cluster.Spec.Metrics.Enabled && len(cluster.Spec.Metrics.ConfigMap) > 0 {
```

Comment:
It is a nit, but personally i prefer `cluster.Spec.Metrics.ConfigMap != ""` because it contains data type information for my human brain :D WDYT?

Reply:
Sure, I don't have a strong opinion on this one. There was just such a convention in my previous org.